subprojects/model-core/src/testFixtures/groovy/org/gradle/internal/reflect/validation/ValidationMessageChecker.groovy

```diff
@@ -35,4 +35,31 @@ trait ValidationMessageChecker {
         String intro = includePropertyName ? "property '$property' " : ""
         "${intro}doesn't have a configured value. This property isn't marked as optional and no value has been configured. Possible solutions: Assign a value to '${property}' or mark property '${property}' as optional. ${learnAt('validation_problems', 'value_not_set')}."
     }
+
+    String methodShouldNotBeAnnotatedMessage(String type, String kind, String method, String annotation, boolean includeDocLink = false) {
+        String message = "Type '$type': $kind '$method()' should not be annotated with: @$annotation. Method '$method' is annotated with an input/output annotation and there is no corresponding getter. Possible solution: Remove the annotations."
+        if (includeDocLink) {
+            message = "${message} ${learnAt('validation_problems', 'ignored_annotations_on_method')}."
+        }
+        message
+    }
+
+    String privateGetterAnnotatedMessage(String property, String annotation, boolean includeType = true, boolean includeLink = false) {
```

Comment:
Can we use `enum`s instead of the booleans? I think that should make it clearer what the booleans actually mean.

Reply:
That would be one enum per parameter (and potentially per method). This is just used in testing, so I'd rather keep it as is for now.

Reply:
I actually think that if we go this route it should be a parameter object rather than enums.